bundles/org.openhab.binding.modbus.stiebeleltron/src/main/java/org/openhab/binding/modbus/stiebeleltron/internal/handler/StiebelEltronHandler.java

```diff
@@ -0,0 +1,736 @@
+        tearDown();
+    }
+
+    /**
+     * Unregister the poll tasks and release the endpoint reference
+     */
+    private void tearDown() {
+        logger.trace("unregisterPollTasks");
+        if (systemInformationPoller != null) {
+            logger.debug("Unregistering systemInformationPoller from ModbusManager");
+            systemInformationPoller.unregisterPollTask();
+
+            systemInformationPoller = null;
+        }
+        if (energyPoller != null) {
+            logger.debug("Unregistering energyPoller from ModbusManager");
+            energyPoller.unregisterPollTask();
+
+            energyPoller = null;
+        }
+        if (systemStatePoller != null) {
+            logger.debug("Unregistering systemStatePoller from ModbusManager");
+            systemStatePoller.unregisterPollTask();
+
+            systemStatePoller = null;
+        }
+        if (systemParameterPoller != null) {
+            logger.debug("Unregistering systemParameterPoller from ModbusManager");
+            systemParameterPoller.unregisterPollTask();
+
+            systemParameterPoller = null;
+        }
+
+        unregisterEndpoint();
+    }
+
+    /**
+     * Returns the current slave id from the bridge
+     */
+    public int getSlaveId() {
+        return slaveId;
+    }
+
+    /**
+     * Get the endpoint handler from the bridge this handler is connected to Checks
+     * that we're connected to the right type of bridge
+     *
+     * @return the endpoint handler or null if the bridge does not exist
+     */
+    private @Nullable ModbusEndpointThingHandler getEndpointThingHandler() {
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            logger.debug("Bridge is null");
+            return null;
+        }
+        if (bridge.getStatus() != ThingStatus.ONLINE) {
+            logger.debug("Bridge is not online");
+            return null;
+        }
+
+        ThingHandler handler = bridge.getHandler();
+        if (handler == null) {
+            logger.debug("Bridge handler is null");
+            return null;
+        }
+
+        if (handler instanceof ModbusEndpointThingHandler) {
+            ModbusEndpointThingHandler slaveEndpoint = (ModbusEndpointThingHandler) handler;
+            return slaveEndpoint;
+        } else {
+            logger.debug("Unexpected bridge handler: {}", handler);
+            throw new IllegalStateException();
+        }
+    }
+
+    /**
+     * Get a reference to the modbus endpoint
+     */
+    private void connectEndpoint() {
+        if (endpoint != null) {
+            return;
+        }
+
+        ModbusEndpointThingHandler slaveEndpointThingHandler = getEndpointThingHandler();
+        if (slaveEndpointThingHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, String.format("Bridge '%s' is offline",
+                    Optional.ofNullable(getBridge()).map(b -> b.getLabel()).orElse("<null>")));
```

Comment:
This is much overhead to work around two null checks. Beside that, there is still a compiler warning left. IMHO you don't need to log the bridge's label, since bridge and things are already correlated in the UI.

Reply:
removed the bridge name